### PR TITLE
added default mkdir mmaps to MoveMapGen

### LINKED ISSUE #560: Feature Request: create mmaps directory with MoveMapGen
**Current behavior**: 
Error no mmap directory
**Expected behavior**: 
Continue 
**Steps to reproduce**:
1.  run MoveMapGen

**Commit hash**: 
https://github.com/vmangos/core/commit/22202bb9263e4a019634a88fdb2e4a7cf90a92c7
**Operating system**: 
Linux 64bit

### PR DESCRIPTION
**Pull request by tonymmm1**
## 🍰 Pullrequest

Added mkdir after directory check in /contrib/mmaps/src/generator.cpp. If there are any errors with my edit please let me know.
This is my first C/C++ contribution to the project and I wrote this quickly so I did not check any styling guides. 

### Issues
- fixes https://github.com/vmangos/core/issues/560

### Todo / Checklist

- [] Feedback/Reviews

### CODE REVIEW

**Comment by ratkosrb** [comment]
You've used tabs, but in this project we use 4 spaces. Please replace all the tabs. Also have you tested if this compiles under windows too? I don't think the CI actually compiles with extractors.

**Comment by tonymmm1** [comment]
@ratkosrb done for the spaces

File: contrib/mmap/src/generator.cpp [hunk 1]
```diff
@@ -41,8 +41,13 @@ bool checkDirectories(bool debugOutput)
     dirFiles.clear();
     if (getDirContents(dirFiles, "mmaps") == LISTFILE_DIRECTORY_NOT_FOUND)
     {
-        printf("'mmaps' directory does not exist, please create it\n");
-        return false;
+        /* create mmaps directory */
+        if (mkdir("mmaps",0777) != 0)
+        {
+            /* return error if operation fails */
+	    printf("'mmaps' directory does not exist, please create it\n");
```
Comment (ratkosrb) [inline contrib/mmap/src/generator.cpp]: this does not line up with the line below

Comment (tonymmm1) [reply contrib/mmap/src/generator.cpp]: I think it was only the printf line

Comment (tonymmm1) [reply contrib/mmap/src/generator.cpp]: @ratkosrb as for the build ci I will look into that

**Comment by tonymmm1** [comment]
@ratkosrb I was also wondering if we should remove travis.yml and the Travis repo hook as the free site will be closing at somepoint and we use Github Actions. As well as removing the badges from the Readme.

**Comment by tonymmm1** [comment]
@ratkosrb Can this be merged if all else looks correct?  Will make extraction easier as I do not need to add additional scripting for directory management.

**Comment by ratkosrb** [comment]
@tonymmm1 The project no longer compiles on Windows! Check the CI, it's failing.

**Comment by GrenderG** [comment]
> @tonymmm1 The project no longer compiles on Windows! Check the CI, it's failing.

Embrace the penguin.

**Comment by tonymmm1** [comment]
@ratkosrb yeah I will look into getting it working cross-platform.

**Comment by tonymmm1** [comment]
Example macro contrib/vmap_extractor/vmapextract/vmapexport.cpp
```
#ifdef _WIN32
#include <Windows.h>
#include <sys/stat.h>
#include <direct.h>
#define mkdir _mkdir
#else
```
 Will do this now.

**Comment by GrenderG** [comment]
> Example macro contrib/vmap_extractor/vmapextract/vmapexport.cpp
> 
> ```
> #ifdef _WIN32
> #include <Windows.h>
> #include <sys/stat.h>
> #include <direct.h>
> #define mkdir _mkdir
> #else
> ```
> 
> Will do this now.

`_mkdir` only takes `dirname` param in Windows.

**Comment by tonymmm1** [comment]
yes I will add a macro for the permissions part following the file I mentioned above as an example.

**Pull request merged**